test/Waives.Http.Tests/ReliableRequestSenderFacts.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using NSubstitute;
+using Polly;
+using Xunit;
+
+namespace Waives.Http.Tests
+{
+    public class ReliableRequestSenderFacts
+    {
+        private readonly RetryLogger _retryLogger;
+        private readonly HttpRequestMessage _request;
+
+        public ReliableRequestSenderFacts()
+        {
+            _retryLogger = new RetryLogger();
+            _request = new HttpRequestMessage(HttpMethod.Get, new Uri("/documents", UriKind.Relative));
+        }
+
+        [Theory]
+        [InlineData(HttpStatusCode.RequestTimeout)]      //408
+        [InlineData(HttpStatusCode.InternalServerError)] //500
+        [InlineData(HttpStatusCode.NotImplemented)]      //501
+        [InlineData(HttpStatusCode.BadGateway)]          //502
+        [InlineData(HttpStatusCode.ServiceUnavailable)]  //503
+        [InlineData(HttpStatusCode.GatewayTimeout)]      //504
+        [InlineData(HttpStatusCode.HttpVersionNotSupported)] //505
+        [InlineData(HttpStatusCode.VariantAlsoNegotiates)]   //506
+        [InlineData(HttpStatusCode.InsufficientStorage)] //507
+        [InlineData(HttpStatusCode.LoopDetected)]        //508
+        [InlineData(HttpStatusCode.NotExtended)]         //509
+        [InlineData(HttpStatusCode.NetworkAuthenticationRequired)] //510
+        public async Task Send_retries_on_error_response(HttpStatusCode statusCode)
+        {
+            var sender = Substitute.For<IHttpRequestSender>();
+            sender
+                .Send(Arg.Any<HttpRequestMessage>())
+                .Returns(
+                    new HttpResponseMessage(statusCode),
+                    Responses.Success());
+
+            var sut = new ReliableRequestSender(
+                _retryLogger.RetryAction,
+                sender);
+
+            await sut.Send(_request);
+
+            Assert.True(_retryLogger.RetryActionCalled);
+        }
+
+        [Theory]
+        [InlineData(HttpStatusCode.Continue)]         //100
```

Comment:
💪 Why was this subset of responses chosen for the test? Should we test all responses not included in the above test?